### PR TITLE
solana-ibc: added `send_packet` method

### PR DESCRIPTION
**Pull request by dhruvja**
Allows us to send packets given that we already have the client,
connection and channels set up.  This is different from
`send_transfer` and no transfer takes place in this method.  It
accepts a packet and stores it.

This method was added in IBC [v0.30.0](https://github.com/cosmos/ibc-rs/blob/8f2ddfefbeb7988bacc27dc6ab64ba7bcd537ce3/CHANGELOG.md?plain=1#L660)

### CODE REVIEW

File: solana/solana-ibc/programs/solana-ibc/src/error.rs [hunk 1]
```diff
@@ -10,9 +10,12 @@ pub(crate) enum Error {
     /// Internal error which ‘should never happen’.
     Internal(&'static str),
 
-    /// Error handling an IBC request.
+    /// Error handling an IBC deliver packet request.
     RouterError(ibc::core::RouterError),
 
+    /// Error handling an IBC send packet request
+    ContextError(ibc::core::ContextError),
+
```
Comment (mina86) [inline solana/solana-ibc/programs/solana-ibc/src/error.rs:18]: Let’s wait for https://github.com/ComposableFi/emulated-light-client/pull/128 which will cover this.

File: solana/solana-ibc/programs/solana-ibc/src/lib.rs [hunk 2]
```diff
@@ -195,6 +196,32 @@ pub mod solana_ibc {
...
+            provable,
+            packets,
+            accounts: Vec::new(), // We are not doing any transfers so no point of having accounts
+            host_head,
+        });
+
+        ibc::core::send_packet(&mut store, packet)
+            .map_err(error::Error::ContextError)
+            .map_err(|err| error!((&err)))?;
+        Ok(())
```
Comment (mina86) [inline solana/solana-ibc/programs/solana-ibc/src/lib.rs:223]:
```suggestion
            .map_err(|err| error!((&err)))
```
should work here, no?

**Review by mina86 — APPROVED** [review]

File: solana/solana-ibc/programs/solana-ibc/src/error.rs (hunk 1 above)
Comment (dhruvja) [reply solana/solana-ibc/programs/solana-ibc/src/error.rs:18]: got it.

File: solana/solana-ibc/programs/solana-ibc/src/lib.rs (hunk 2 above)
Comment (dhruvja) [reply solana/solana-ibc/programs/solana-ibc/src/lib.rs:223]: Yes yes, the above would work.

Comment (dhruvja) [reply solana/solana-ibc/programs/solana-ibc/src/lib.rs]: Nvm, saw that its already been added to master with #128 .

**Review by mina86 — APPROVED** [review]

**Pull request merged**